### ISSUE TITLE
Support Transformers.js

### ISSUE THREAD

**Issue by jens-f** [promptfoo/promptfoo#2056]
Would you consider supporting [Transformers.js](https://github.com/huggingface/transformers.js) as an additional provider? This library provides access to text generation and content classification functionality using many different models. With the new V3 the support grew even further. In addition, it might be the most convenient way to create text embeddings locally since it would run inside the node application and hence no external server needs to be started.

**Comment by mldangelo** [comment]
Hi @jens-ghc, yes, this is something we would be interested in! Would you be open to contributing to the integration? I'd be happy to help on a call. For what it's worth, we also currently have a [Hugging Face integration](https://www.promptfoo.dev/docs/providers/huggingface/) but would love to revisit and improve it alongside a potential transformers.js integration.

**Comment by jens-ghc** [comment]
Hi @mldangelo Let me look into it. At least the embeddings-part might be straight forward. One question: some embeddings models require a "prefix" to be set in order for them to work properly, is this something that promptfoo supports through configuration files?

**Comment by mldangelo** [comment]
Yes! Take a look at https://github.com/promptfoo/promptfoo/blob/main/src/providers.ts (the entrypoint) and then any of our specific provider implementations https://github.com/promptfoo/promptfoo/tree/main/src/providers.
